### PR TITLE
Fixed compiler warnings I.

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Signed-off-by: Tomas Kraus <tomas.kraus@oracle.com>

### CODE REVIEW

File: fastinfoset/src/main/java/com/sun/xml/fastinfoset/Decoder.java [hunk 1]
```diff
@@ -1938,28 +1954,20 @@ protected final boolean _isFastInfosetDocument() throws IOException {
                 _octetBuffer[3] != EncodingConstants.BINARY_HEADER[3]) {
             
             // Check for each form of XML declaration
-            for (int i = 0; i < EncodingConstants.XML_DECLARATION_VALUES.length; i++) {
-                _octetBufferLength = EncodingConstants.XML_DECLARATION_VALUES[i].length - _octetBufferOffset;
+            for (byte[] XML_DECLARATION_VALUES : EncodingConstants.XML_DECLARATION_VALUES) {
```
Comment (lukasj) [inline fastinfoset/src/main/java/com/sun/xml/fastinfoset/Decoder.java:1957]: use lower case for non-static variable name

File: fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java [hunk 2]
```diff
@@ -735,15 +774,15 @@ protected final void encodeCharacters(char[] ch, int offset, int length) throws
     }
 
     /**
-     * Encode a chunk of Character Information Items.
+     * Encode a chunk of Character Information Items.If the array of characters is to be indexed (as determined by
+    {@link #isCharacterContentChunkLengthMatchesLimit(int)} then the array is not cloned
+ when adding the array to the vocabulary.
```
Comment (lukasj) [inline fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java:779]: fix formatting

File: fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java [hunk 3]
```diff
@@ -735,15 +774,15 @@ protected final void encodeCharacters(char[] ch, int offset, int length) throws
...
      *
-     * If the array of characters is to be indexed (as determined by
-     * {@link #isCharacterContentChunkLengthMatchesLimit(int) } then the array is not cloned
-     * when adding the array to the vocabulary.
      *
      * @param ch the array of characters.
      * @param offset the offset into the array of characters.
      * @param length the length of characters.
+     * @throws java.io.IOException
```
Comment (lukasj) [inline fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java:785]: I'd add "for errors"; the way it is done just turns one warning into another

File: fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java [hunk 4]
```diff
@@ -948,15 +997,15 @@ protected final void encodeComment(char[] ch, int offset, int length) throws IOE
     }
 
     /**
-     * Encode a Comment Information Item.
+     * Encode a Comment Information Item.If the array of characters that is a comment is to be indexed (as
+ determined by {@link #isCharacterContentChunkLengthMatchesLimit(int)}) then
+ the array is not cloned when adding the array to the vocabulary.
```
Comment (lukasj) [inline fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java:1002]: formatting

File: fastinfoset/src/main/java/com/sun/xml/fastinfoset/Decoder.java (hunk 1 above)
Comment (Tomas-Kraus) [reply fastinfoset/src/main/java/com/sun/xml/fastinfoset/Decoder.java:1958]: Yes, looks like this was caused by automatic Netbeans changes.

File: fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java (hunk 3 above)
Comment (Tomas-Kraus) [reply fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java:785]: OK, fixed.

File: fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java (hunk 4 above)
Comment (Tomas-Kraus) [reply fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java:1002]: Fixed, another javadoc broken by NetBeans.

File: fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java (hunk 2 above)
Comment (Tomas-Kraus) [reply fastinfoset/src/main/java/com/sun/xml/fastinfoset/Encoder.java]: Fixed.

**Review by lukasj — APPROVED** [review]
LGTM

**Pull request merged**